16-projects.Rmd

```diff
@@ -265,6 +265,28 @@ Note that you need to change the vignette title in both the `title` field and th
 
 The vignette output format does not have to be HTML. It can also be PDF, so you can use `output: pdf_document`, too. Any other output formats that create HTML or PDF are also okay, such as `beamer_presentation` and `tufte::tufte_html`. However, currently R only recognizes HTML and PDF vignettes.
 
+## R package templates
+
+In section \@ref(package-vignette), you learned how to include an R Markdown document in a package as a vignette. This is an excellent way to demonstrate the package's functionality, but it is not the only way that R Markdown documents can be used in an R package. As a package developer, you may also include editable R Markdown templates in your package.
+
+R Markdown templates help package users interact with a package by provide boilerplate code or document structure. In fact, section \@ref(package-vignette) on package vignettes provides a compelling use case. Figure \@ref(fig:package-vignette) shows how RStudio users can use the menu `File -> New File -> R Markdown -> From Template` to retrieve the "Package Vignette (HTML)" template from the **rmarkdown** package. This template offers a file to edit which already contains the appropriate YAML metadata for a package vignette. 
+
+Templates can also include boilerplate content and showcase a package's features. For example, the **flexdashboard** package contains a sample dashboard and the **xaringan** package contains a presentation template which demonstrates many formatting options.
+
+To add an R Markdown template to a package, create a subdirectory of the `inst/rmarkdown/templates` directory. Within this directory, you will save at least two files:
+
+1. A file named `template.yaml` which gives the RStudio IDE basic metadata such as a human-readable name for the template
+
+```yaml
+name: Example Template
+```
+
+2. An R Markdown document saved under `skeleton/skeleton.Rmd` which contains the template. This may contain anything you wish to put in an R Markdown document. 
+
+Optionally, the `skeleton` folder may also include additional resources like style sheets or images used by your template. 
```

Comment:
In this case, it would be copied in the directory where the Rmd file is created, unless `create_dir: true` is in the yaml, and the IDE, will have a box to create a new folder to contain all the files.
See examples: 
* with `create_dir: true`; https://github.com/rstudio/learnr/blob/master/inst/rmarkdown/templates/tutorial/template.yaml
* with `create_dir: false` : https://github.com/rstudio/flexdashboard/blob/master/inst/rmarkdown/templates/flex_dashboard/template.yaml